### DISCUSSION TITLE
gongju/junit

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#170] [Announcements]
# gongju/junit

01、前世今生 你好呀，我是 JUnit，一个开源的 Java 单元测试框架。在了解我之前，先来了解一下什么是单元测试。单元测试，就是针对最小的功能单元编写测试代码。在 Java 中，最小的功能单元就是方法，因此，对 Java 程序员进行单元测试实际上就是对 Java 方法的测试。 为什么要进行单元测试呢？因为单元测试可以确保你编写的代码是符合软件需求和遵循开发规范的。单元测试是所有测试中最底层的一类测试，是第一个环节，也是最重要的一个环节，是唯一一次能够达到代码覆盖率 100% 的测试，是整个软件测试过程的基础和前提。可以这么说，单元测试的性价比是最好的。

https://tobebetterjavaer.com/gongju/junit.html

**Comment by mengfh** [comment]
1）单元测试的代码本身必须非常名单明了，能一下看明白，决不能再为测试代码编写测试代码。：：：@@非常名单明
